lib/datasets/wikipedia-kyoto-japanese-english.rb

```diff
@@ -0,0 +1,215 @@
+require 'csv'
+require "rexml/streamlistener"
+require "rexml/parsers/baseparser"
+require "rexml/parsers/streamparser"
+require "time"
+
+require_relative "dataset"
+require_relative "tar-gz-readable"
+
+module Datasets
+  class WikipediaKyotoJapaneseEnglish < Dataset
+    include TarGzReadable
+
+    Article = Struct.new(:source,
+                         :copyright,
+                         :contents,
+                         :sections)
+
+    Section = Struct.new(:id,
+                         :title,
+                         :contents)
+
+    class Title < Struct.new(:section,
+                             :japanese,
+                             :english)
+      def title?
+        true
+      end
+
+      def sentence?
+        false
+      end
+    end
+
+    Paragraph = Struct.new(:id,
+                           :sentences)
+
+    class Sentence < Struct.new(:id,
+                                :section,
+                                :paragraph,
+                                :japanese,
+                                :english)
+      def title?
+        false
+      end
+
+      def sentence?
+        true
+      end
+    end
+
+    Entry = Struct.new(:japanese,
+                         :english)
+
+    def initialize(type: :article)
+      unless [:article, :lexicon].include?(type)
+        raise ArgumentError, "Please set type :article or :lexicon #{type.inspect}"
+      end
+
+      super()
+      @type = type
+      @metadata.id = "wikipedia-kyoto-japanese-english"
+      @metadata.name =
+        "The Japanese-English Bilingual Corpus of Wikipedia's Kyoto Articles"
+      @metadata.url = "https://alaginrc.nict.go.jp/WikiCorpus/index_E.html"
+      @metadata.licenses = ["CC-BY-SA-3.0"]
+      @metadata.description = <<-DESCRIPTION
+"The Japanese-English Bilingual Corpus of Wikipedia's Kyoto Articles"
+aims mainly at supporting research and development relevant to
+high-performance multilingual machine translation, information
+extraction, and other language processing technologies. The National
+Institute of Information and Communications Technology (NICT) has
+created this corpus by manually translating Japanese Wikipedia
+articles (related to Kyoto) into English.
+      DESCRIPTION
+    end
+
+    def each(&block)
+      return to_enum(__method__) unless block_given?
+
+      data_path = download_tar_gz
+
+      open_tar_gz(data_path) do |tar|
+        tar.each do |entry|
+          next unless entry.file?
+          base_name = File.basename(entry.full_name)
+          case @type
+          when :article
+            next if base_name == "kyoto_lexicon.csv" || base_name == "readme.pdf" || base_name == "Wiki_Corpus_List_2.01.csv"
```

Comment:
How about checking `.xml` extension?

```suggestion
            next unless base_name.end_with?(".xml")
```

Reply:
I think your idea is better. Thank you.